packaging/aznfs/DEBIAN/prerm

```diff
@@ -26,7 +26,25 @@ if [ "$1" == 'remove' -o "$1" == 'purge' ]; then
 	fi
 fi
 
-# Stop aznfswatchdog service and clean it.
-systemctl stop aznfswatchdog
-systemctl disable aznfswatchdog
-echo "Stopped aznfswatchdog service."
\ No newline at end of file
+flag_file="/tmp/.update_in_progress_from_watchdog.flag"
+
+if [ -f "$flag_file" ]; then
+	# Get the PID of aznfswatchdog
+	aznfswatchdog_pid=$(pgrep aznfswatchdog)
+	
+	# Read the PID from the flag file
+	aznfswatchdog_pid_inside_flag=$(cat $flag_file)
+	
+	if [ "$aznfswatchdog_pid" != "$aznfswatchdog_pid_inside_flag" ]; then
+		# The flag file is stale, remove it
+		rm -f "$flag_file"
+		echo "Removed stale flag file."
+	fi
+fi
+
+if [ ! -f "$flag_file" ]; then
```

Comment:
This is else condition of above if condition, right? Then write it in proper if else.